Added ContainerID and ContainerID2 for chest subtypes

JeseGamerHD opened the pull request:
### What is the new feature?
I got tired of checking the internal id for different chest subtypes so I created an ID. This should also be a bit clearer compared to a number: `ContainerID.FrozenChest` versus using just `11`. Since there are two containers `TileID.Containers` and `TileID.Containers2` I made two classes: `ContainerID` and `ContainerID2`.
### Why should this be part of tModLoader?
To reduce potential confusion with "magical numbers". Also not having to look up the internal id from wiki or from the spritesheets is nice.
### Are there alternative designs?
The naming could maybe be something other than ContainerID like ChestID for example. Other than that I'm not sure.
### Sample usage for the new feature
Checking if a tile is a Sandstone Chest
`if (chestTile.TileType == TileID.Containers2 && chestTile.TileFrameX == ContainerID2.SandstoneChest * 36) { }`

### ExampleMod updates
Changed `ChestItemWorldGen.cs` to use the new ID and updated the comments.

Commit:
Added ContainerID and ContainerID2 for chest subtypes

JavidPack commented on patches/tModLoader/Terraria/ID/ContainerID.cs:
```diff
@@ -0,0 +1,83 @@
+	public const int GolfChest = 9;
+	public const int SandstoneChest = 10;
+	public const int BambooChest = 11;
+	public const int DesertChest = 12;
```
This won't compile. Typo? (DesertChest defined twice.)

JeseGamerHD replied:
Forgot to add Locked in front of it (fixed now). I guess I didn't open the project correctly since the IDE didn't complain. Also, would it be better to multiply the values by 36 (chest width) in advance in the ID classes?

JavidPack replied:
No, these represent the styles, not the coordinates of the top left tile.

JeseGamerHD committed:
Update ContainerID.cs

Typo

JeseGamerHD committed:
GoldenPirateChest

Forgot that Golden Chest and the other version from pirates had a same name, now fixed

JeseGamerHD committed:
Merge branch 'tModLoader:1.4.4' into 1.4.4
